### ISSUE TITLE
Code Coverage using angr

### ISSUE THREAD

**Issue by Kigorky** [angr/angr#3397]
Hi angr group,

i need your help in order to perform code coverage by using angr. Given a branch condition 
```
if(condition)
     .....
 else    
     .....
```

 I 'm trying to retrieve N inputs that end up in the if branch, N inputs for else branch and so on for the other branches of the program:
First of all i wrote the following very basic angr script: 

```
def myfunc():
    p = angr.Project('mytest', main_opts={'backend':'elf', 'arch': 'armhf', 'base_addr': 0x0}, auto_load_libs=False)
    input = claripy.BVS('input', 64)
    init_state = p.factory.full_init_state(
          args=['./mytest'],
          stdin=input,
    )
    
    sm = p.factory.simgr(init_state)
    sm.explore(find=0x00010b9c)
    found = sm.found[0]
    solution = found.solver.eval_upto(input, 10, cast_to=bytes)
    print(repr(solution))
   # open("out_"+sys.argv[1],"wb").write(sm.found[0].posix.dumps(0))


if __name__ == '__main__': 
    myfunc();
```

That one seems to be working and i got the following output:
`[b'\x00\x00\x00\x00\x00\x00\x00\x1c', b'\xf5\xf5\xf5\xf5\xf5\xf5\xf5\x00', b'\x00\x00\x00\x00\x00\x00\x00\x0c', b'\x00\x00\x00\x00\x00\x00\x00<', b'\x00\x00\x00\x00\x00\x00\x00|', b'\x00\x00\x00\x00\x00\x00\x03\xfc', b'\x00\x00\x00\x00\x00\x00\x00\x04', b'\x00\x00\x00\x00\x00\x00\x00\xfc', b'\x00\x00\x00\x00\x00\x00\x00\x00', b'\x00\x00\x00\x00\x00\x00\x01\xfc']`

Can you please validate the methodology for me? Is that the right way to use angr in order to retrieve N inputs which exactly end up in that branch ?

Thank you so much

**Comment by rhelmot** [comment]
Yes, `eval_upto` on a single state is the right way to retrieve N inputs which will drive the program down the same branch as the simulated state.
